build-system/tasks/e2e/mocha-custom-json-reporter.js

```diff
@@ -27,9 +27,15 @@ const {
 const {Base} = require('mocha').reporters;
 const {inherits} = require('mocha').utils;
 
-async function writeOutput(output, filename) {
+function writeOutput(output, filename) {
   try {
-    await fs.writeFile(filename, JSON.stringify(output, null, 4));
+    fs.writeFileSync(filename, JSON.stringify(output, null, 4));
```

Comment:
Using sync here because this happens at the end & shouldn't block too much. For a bit of support, there's a widely used reporter that also writes to a file synchronously -- https://www.npmjs.com/package/mocha-junit-reporter. See https://github.com/michaelleeallen/mocha-junit-reporter/blob/master/index.js#L471